File: src/main/java/models/entities/Address.java
package models.entities;

import com.fasterxml.uuid.Generators;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Address {
	/* UUID */
    private String id = Generators.timeBasedGenerator().generate().toString();

	private String street;
	private String number;
	private String complement;
	private String neighborhood;
	private String zipCode;
	private String city;
	private String state;
	private String country;
}

File: frontend/app/validations/address.js
import {
  validatePresence,
  validateLength,
  validateFormat
} from 'ember-changeset-validations/validators';

export default {
  zipCode: [
    validateFormat({ regex: /[0-9]{5}-[0-9]{3}/ })
  ],
  street: [
    validatePresence(true),
    validateLength({ min: 4 })
  ],
  number: [
    validatePresence(true)
  ],
  neighborhood: [
    validatePresence(true)
  ],
  city: [
    validatePresence(true)
  ],
  state: [
    validatePresence(true)
  ],
};
